src/Agent.Listener/JobDispatcher.cs

```diff
@@ -29,45 +29,62 @@ public sealed class JobDispatcher : AgentService, IJobDispatcher
             using (var processChannel = HostContext.CreateService<IProcessChannel>())
             using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
             {
```

Comment:
I didn't change much in this file. I studied this file a bit, and made some readability improvements for the stumbling blocks that I hit while reading the file (added comments, named args, deleted unnecessary code).

I also added some TODOs for things that appear to need further thought. I think Ting has some of these in mind already. I just wanted to call them out explicitly so we don't lose track.